modules/missionmail.js

```diff
@@ -0,0 +1,258 @@
+  // /On\s.+wrote:\s*/im,
+  /^>.*$/m,
+  /^-+ original message -+$/m,
+  /^from:.*$/m,
+  /^sent:.*$/m,
+  /^to:.*$/m,
+  /^subject:.*$/m,
+  /^date:.*$/m
+];
+/** @type {receive.ImapFlow | undefined} */
+let receiver;
+/** @type {Map<string, {approve:() => any, reject:() => any}>} */
+const awaitingMods = new Map();
+async function logNPull() {
+  const creds = c.google.mail;
+  try {
+    receiver?.removeAllListeners();
+    receiver?.close();
+    receiver = new receive.ImapFlow({
+      auth: {
+        user: creds.email,
+        pass: creds.gAccountAppPass
+      },
+      host: 'imap.gmail.com',
+      port: 993,
+    });
+    await receiver.connect();
+    await receiver.mailboxOpen("INBOX");
+    sendUnsent();
+    // this shouldn't run too often, but it could theoretically depending on how much the email server spams.
+    receiver.on("exists", sendUnsent);
+  } catch (error) {
+    u.errorHandler(error, `login & pull missionmail`);
+    receiver = undefined;
+  }
+}
+async function sendUnsent() {
+  if (receiver?.usable) {
+    try {
+      let messageIds = (await receiver.search({ unKeyword: 'icarusForwarded' }));// , from: "*@missionary.org" }))
+      messageIds = messageIds.filter(msgId => {
+        return awaitingMods.has(msgId + "") ? undefined : msgId;
+      });
+
+      const messages = await receiver.fetchAll(messageIds, { source: true });
+      // console.log(messages);
+      // console.log("Checking for new emails:", messages?.length);
+      for (const rawMsg of messages) {
+        // parse the email source into readable stuff
+        const parsed = await new Promise((resolve) => {
+          const parser = new interpret.MailParser();
+          parser.on("end", result => resolve(result));
+          parser.write(rawMsg.source);
+          parser.end();
+        });
+        // make sure there is text
+        if (!parsed.text || parsed.text.length < 1) {
+          if (parsed.html) {
+            parsed.text = htmlparse.convert(parsed.html);
+          } else {
+            parsed.text = parsed.subject;
+          }
+        }
+        if (!parsed.text) {throw new Error("unable to parse email with no discrnable text, html, or subject");}
+        // trim the reply quote from the bottom if there is one (for some reason it was bypassing email replace)
+        for (const regex of replyRegexes) {
+          const match = parsed.text.toLowerCase().search(regex);
+          if (match && match > 0) {
+            // parsed.fullText = parsed.text;
+            parsed.text = parsed.text.substring(0, match);
+            break; // Stop after the first match to avoid over-trimming
+          }
+        }
+        parsed.text = parsed.text.trimEnd();
+        // search for profanity
+        const pfViolations = pf.scan(parsed.text + "");
+        const bannedViolations = [banned.links, banned.words, banned.scam].flat().filter(bannedString => parsed.text.includes(bannedString) ? bannedString : undefined);
```

Comment:
If the email contains profanity, it'll be handled by 01-filter.js. No need for additional filtering here.